docs/monthly/2023-05-monthly.md

```diff
@@ -0,0 +1,250 @@
+more than one branch to merge... you can always feel it, don't you?
+
+So let's see what's going on there and let's take a look at this run from [may
+the 10th](https://github.com/input-output-hk/hydra/actions/runs/4933005294):
+- Building and testing takes 19 minutes for the longest
+- Generating haddock documentation and running the benchmarks takes 28 minutes
+  for the longest
+- documentation (which will need artifact generated in previous steps) will take
+  14 minutes
+- In total, this run took 1 hour and 16 minutes.
+
+Our first focus has been on the build and test stage. We're expecting the
+plutus-merkle-tree to run fast but [it took 8 minutes and 52
+seconds](https://github.com/input-output-hk/hydra/actions/runs/4933005294/jobs/8816564512)
+and if we look in detail, we can see that 7m and 11 seconds have been spent
+setting up the build environment. Stated otherwise, 81% of the build time is
+downloading binary dependencies from some nix cache. How can we reduce the size
+of our dependencies?
+
+If we want to compile our code and run the test, we need cabal and other nix
+dependencies that will download _the world_ before doing anything. But what if
+we decide to, straight on, run the test with nix and let it decide what it needs
+to compile because, maybe, most of our code did not change, the test binary is
+already available in some cache and we just run it!
+
+This what we did in [#867](https://github.com/input-output-hk/hydra/pull/867).
+Merging this P.R. the [build on master took 45
+minutes](https://github.com/input-output-hk/hydra/actions/runs/5003046049) and,
+specifically, the same [plutus-merkle-tree
+job](https://github.com/input-output-hk/hydra/actions/runs/5003046049/jobs/8963773583)
+only took 1 minute and 44 seconds, only 20% of the time observed before.
+
+Then we decided that we wanted to spend less time on documentation (14 minutes,
+remember). It happened that most of the time spent by this process is web-site
+optimization. That's fine for master because we want to publish an optimized
+web-site. But what about all the builds in branches for which the web-site will
+never be published? We need to keep this documentation step in branches because
+it also provide us security by checking that we didn't introduce too much mess
+in the doc, like broken links, for instance. But we definitely got rid of the
+optmization step.
+
+That is what we did in [#880](https://github.com/input-output-hk/hydra/pull/880)
+and it helped up [save 10 minutes from this
+step](https://github.com/input-output-hk/hydra/actions/runs/5067084637/jobs/9098252031).
+
+Our first goal was to reduce coninuous integration execution time when pushing
+on branches and this has been improved. We're now having execution time
+significantly under 30 minutes where it used to be 45 minutes or event an hour.
+
+We had some issue with compilation output, swallowed by nix, which have been
```

Comment:
I suggest replacing 'swallowed' with something else too, unless it is commonly used in development language??